dcs/action.py

```diff
@@ -1787,6 +1787,166 @@ def dict(self):
+
+    @classmethod
+    def create_from_dict(cls, d, mission):
+        return cls(d["coalitionlist"], ResourceKey(d["file"]), d["seconds"], d["clearview"], d["start_delay"],
+                   d["horzAlignment"], d["vertAlignment"], d["size"], d["size_units"])
+
+    def dict(self):
+        d = super().dict()
+        d["coalitionlist"] = self.coalition
+        return d
+
+
+class PictureToCountry(PictureAction):
+    predicate = "a_out_picture_c"
+
+    def __init__(self, country: int, file_res_key: ResourceKey, seconds: int,
+                 clearview: bool, start_delay: int,
+                 horz_alignment: str, vert_alignment: str,
+                 size: int, size_units: str):
+        super().__init__(predicate=self.predicate, file_res_key=file_res_key,
+                         seconds=seconds, clearview=clearview,
+                         start_delay=start_delay, horz_alignment=horz_alignment,
+                         vert_alignment=horz_alignment, size=size, size_units=size_units)
+        self.country = country
+        self.params = [self.country, self.file_res_key, seconds, clearview,
+                       start_delay, horz_alignment, vert_alignment, size, size_units]
+
+    @classmethod
+    def create_from_dict(cls, d, mission):
+        return cls(d["countrylist"], ResourceKey(d["file"]), d["seconds"], d["clearview"], d["start_delay"],
+                   d["horzAlignment"], d["vertAlignment"], d["size"], d["size_units"])
+
+    def dict(self):
+        d = super().dict()
+        d["countrylist"] = self.country
+        return d
+
+
+class PictureToGroup(PictureAction):
+    predicate = "a_out_picture_g"
+
+    def __init__(self, group: int, file_res_key: ResourceKey, seconds: int,
+                 clearview: bool, start_delay: int,
+                 horz_alignment: str, vert_alignment: str,
+                 size: int, size_units: str):
+        super().__init__(predicate=self.predicate, file_res_key=file_res_key,
+                         seconds=seconds, clearview=clearview,
+                         start_delay=start_delay, horz_alignment=horz_alignment,
+                         vert_alignment=horz_alignment, size=size, size_units=size_units)
+        self.group = group
+        self.params = [self.group, self.file_res_key, seconds, clearview,
+                       start_delay, horz_alignment, vert_alignment, size, size_units]
+
+    @classmethod
+    def create_from_dict(cls, d, mission):
+        return cls(d["group"], ResourceKey(d["file"]), d["seconds"], d["clearview"], d["start_delay"],
+                   d["horzAlignment"], d["vertAlignment"], d["size"], d["size_units"])
+
+    def dict(self):
+        d = super().dict()
+        d["group"] = self.group
+        return d
+
+
+class PictureToUnit(PictureAction):
+    predicate = "a_out_picture_u"
+
+    def __init__(self, unit: int, file_res_key: ResourceKey, seconds: int,
```

Comment:
Same as the previous two, but for `unit`

Reply:
This is now unit_id -- if you can help me solve circular import, I don't mind changing it to proper types.

Reply:
@DanAlbert - unit_id to Unit can't be done without refactoring the code which parses triggers. At the moment trigger/action parsing code is unaware of the context of coalition/country it's parsing for, therefore unit_id, since it's not unique, cannot be resolved simply. I'd rather keep this as is and refactor the code later that focuses on adding context to trigger/action parsers, then refactor this piece to use it. Thoughts?